### PR TITLE
feat: improve form-field functionality

### LINKED ISSUE #160: refactor(field): compare with mat-form-field
Currently whenever an error message is shown, everything below it will be moved, due to a new element being added.

One solution would be to reserve space below the input field, so one single line error message could be shown, without it moving everything below.
This would mean having always some white space reserved in sbb-field.

Also refactor sbb-field to sbb-form-field.

### LINKED ISSUE #489: refactor: Deprecate size input on sbb-field
DoD:
- mode attribute of sbb-field is deprecated
- css classes for sizing as alternative are provided
- examples and documentation updated

### PR DESCRIPTION
**Pull request by jeripeierSBB**
This PR refactors the field module to the form-field module. It improves the internal
functionality in regards to connection to the input and accessibility.

BREAKING CHANGE: The field module has been majorly refactored. The `sbb-field`
has been renamed to `sbb-form-field` and now requires a compatible form control
(`<input sbbInput ...>`, `<select sbbInput ...>`, `<textarea sbbInput>`, `<sbb-select>`).
The `sbb-form-field` [mode] input has been deprecated and replaced with appropriate
css classes (`.sbb-form-field-short`, `.sbb-form-field-medium` and `.sbb-form-field-long`).
The `sbb-form-error` has been renamed to `sbb-error`.
The `sbb-label` [for] input has been deprecated, as it is no longer in use, as that
functionality is handled internally inside `sbb-form-field`.
The `sbb-form-field` now has a default width, which is consistent for all supported
form controls.

Closes #160, closes #489

### CODE REVIEW

File: src/angular-public/datepicker/datepicker.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The date input supports min and max dates. The datepicker emits events for openi
 The initial focus for the datepicker is the current date value.
 
 ```html
-<sbb-field mode="long">
+<sbb-form-field mode="long">
```
Comment (jeripeierSBB) [inline src/angular-public/datepicker/datepicker.md:34]: remove mode

File: src/angular-public/datepicker/datepicker.md [hunk 2]
```diff
@@ -56,17 +56,17 @@ manual min has been set.
 
 ```html
 <form [formGroup]="twoDatepickersForm">
-  <sbb-field mode="medium">
+  <sbb-form-field mode="medium">
```
Comment (jeripeierSBB) [inline src/angular-public/datepicker/datepicker.md:59]: mode

File: src/angular-public/datepicker/datepicker.md [hunk 3]
```diff
@@ -56,17 +56,17 @@ manual min has been set.
...
     <sbb-label for="Datum">Datumsbereich innerhalb eines Monats</sbb-label>
     <sbb-datepicker [connected]="second" arrows>
       <input sbbDateInput sbbInput formControlName="firstDatepicker" />
     </sbb-datepicker>
-  </sbb-field>
-  <sbb-field mode="medium">
+  </sbb-form-field>
+  <sbb-form-field mode="medium">
```
Comment (jeripeierSBB) [inline src/angular-public/datepicker/datepicker.md:65]: mode

File: src/angular-public/form-field/form-field.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    // TODO(crisbeto): we need this workaround in order to support both Ivy and ViewEngine.
+    //  We should clean this up once Ivy is the default renderer.
+    return this._explicitFormFieldControl || this._controlNonStatic || this._controlStatic;
+  }
+  set _control(value) {
+    this._explicitFormFieldControl = value;
+  }
+  private _explicitFormFieldControl: SbbFormFieldControl<any>;
+
+  @ContentChild(SbbLabel) _labelChildNonStatic: SbbLabel;
```
Comment (jeripeierSBB) [inline src/angular-public/form-field/form-field.ts:103]: can be private, not? (and following 2 lines)

File: src/angular-public/lightbox/lightbox.md [hunk 5]
```diff
@@ -6,9 +6,9 @@ The lightbox can be used to open modal dialogs as see below
   <div class="sbbsc-lightbox-example-1-content">
     <h1 sbbLightboxTitle>Hi {{data.name}}</h1>
     <p>What's your favorite animal?</p>
-    <sbb-field mode="long">
+    <sbb-form-field mode="long">
```
Comment (jeripeierSBB) [inline src/angular-public/lightbox/lightbox.md:9]: mode

File: src/angular-public/select/select.md [hunk 6]
```diff
@@ -7,22 +7,22 @@ user.
 You can use select component with sbb style using `<sbb-select>` and `<sbb-option>` as seen below
 
 ```html
-<sbb-field mode="long">
+<sbb-form-field mode="long">
```
Comment (jeripeierSBB) [inline src/angular-public/select/select.md:10]: mode

File: src/angular-public/select/select.md [hunk 7]
```diff
@@ -33,7 +33,7 @@ multiple values at once.
 
 ```html
 <h4>Multiple Example</h4>
-<sbb-field mode="long">
+<sbb-form-field mode="long">
```
Comment (jeripeierSBB) [inline src/angular-public/select/select.md:36]: mode

Comment (jeripeierSBB) [reply src/angular-public/select/select.md:36]: a few times more in this file, I won't comment anymore :)

File: package.json [hunk 8]
```diff
@@ -8,6 +8,7 @@
   },
   "scripts": {
     "postinstall": "node tools/postinstall/apply-patches.js && ngcc && yarn -s build:public2business",
+    "p2b": "ng g .:public2business",
     "build": "ts-node-script ./scripts/build.ts",
     "build:public2business": "ng g .:public2business",
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply package.json:13]: Yes, it's a convenience

File: src/angular-public/form-field/form-field.ts (hunk 4 above)
Comment (kyubisation) [reply src/angular-public/form-field/form-field.ts:103]: I don't think that this would be compatible for ViewEngine.

**Pull request merged**